test/functional/api/cas/casadm.py

```diff
@@ -336,6 +387,17 @@ def remove_core(cache_id: int, core_id: int, force: bool = False, shortcut: bool
     )
     if output.exit_code != 0:
         raise CmdException("Failed to remove core.", output)
+
+    core_to_remove = next(
+        iter(
+            [
+                core
+                for core in TestRun.dut.core_list
+                if core.cache_id == cache_id and core.core_id == core_id
+            ]
+        )
+    )
+    TestRun.dut.core_list.remove(core_to_remove)
```

Comment:
You can probably do it like you did stopping cache:
TestRun.dut.core_list = [core for core in TestRun.dut.core_list if core.cache_id != cache_id or core.core_id != core_id ]